File: chain-of-responsibility/chain.js
class Aprobador {
    constructor(nombre, limite, siguiente = null) {
        this.nombre = nombre;
        this.limite = limite;
        this.siguiente = siguiente;
    }

    aprobar(monto) {
        if (monto <= this.limite) {
            console.log(`${this.nombre} aprueba`);
        } else if (this.siguiente) {
            this.siguiente.aprobar(monto);
        }
    }
}

File: chain-of-responsibility/chain.py
class Aprobador:
    def __init__(self, nombre, limite, siguiente=None):
        self.nombre = nombre
        self.limite = limite
        self.siguiente = siguiente

    def aprobar(self, monto):
        if monto <= self.limite:
            print(f"{self.nombre} aprueba")
        elif self.siguiente:
            self.siguiente.aprobar(monto)
